### ISSUE TITLE
Chemical Substance (should this work??) - Gene KCNMA1 (HGNC:6284) 

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/testing#71]
**Query:**[ kcnma1.json](https://github.com/NCATSTranslator/testing/blob/main/ars-requests/not-none/1.1/ChemicalSubstanceKCNMA1.json)
- we're pretty sure this should not work since ChemicalSubstance is not in Biolink 2.0 but there are results
- we'd like to understand how and when we can ask these types of TRAPIs

**PK:** 324c4934-0e68-4866-b33f-f9072ce39967
**HGNC:** 6284
**Control:** maybe not control, but PMI Relay Use Case suggests Lisdexamfetamine (vyvanse), methamphetamines
**[Results Tracking Sheet](https://docs.google.com/spreadsheets/d/1Y2yfUFl2SIgv5KkyxdQB94UE3HtVdJNJ7KdnwQgZcYI/edit?usp=sharing)**

![image](https://user-images.githubusercontent.com/38321826/120731399-8dab4780-c4b1-11eb-9db1-8dec78ffe891.png)

**Comment by andrewsu** [comment]
using curl, BTE retrieves 136 results (including methamphetamine but not Lisdexamfetamine / vyvanse) 
```
curl --silent --location --request POST "https://api.bte.ncats.io/v1/query" --header "Content-Type: application/json" --data-raw "{
    \"message\": {
        \"query_graph\": {
            \"nodes\": {
                \"n0\": {
                    \"categories\": [
                        \"biolink:ChemicalSubstance\"
                    ]
                },
                \"n1\": {
                    \"ids\": [
                        \"HGNC:6284\"
                    ],
                    \"categories\":[\"biolink:Gene\"]
                }
            },
            \"edges\": {
                \"e0\": {
                    \"subject\": \"n0\",
                    \"object\": \"n1\"
                }
            }
        }
    }
}"  | jq '.message.results | length'
```

Anyone have any idea what it means when the results are n/a but the TRAPI validates successfully?  @dkoslicki I can't find any more detailed error message through ARAX -- am I missing it anywhere?  @MarkDWilliams, I was expecting to be able to see more from ARS at https://ars.transltr.io/ars/api/messages/ec030f44-d4d0-468c-b608-72226658e5a0, but that gives me a 500 error.  Any pointers?

**Comment by edeutsch** [comment]
> Anyone have any idea what it means when the results are n/a but the TRAPI validates successfully?

A valid TRAPI message does not require a results[] list. If the response that comes back is valid TRAPI but contains no results[] list, then this is the situation.

> I can't find any more detailed error message through ARAX

I don't quite understand what you mean by more detailed error message through ARAX?

After clicking on a particular response, you can click on [Dev Info] in the lower left to see the TRAPI that came back from the ARS (augmented with validator failure information if it failed)

**Comment by dkoslicki** [comment]
ARAX does produce results, but an error in the CHP KP stops the results from being returned. This has now been fixed in [this ARAX issue](https://github.com/RTXteam/RTX/issues/1500) and will be deployed soon.

**Comment by edeutsch** [comment]
ARAX fix is now deployed. If ARS tries again, it should work.

I'm confused by this:
> we're pretty sure this should not work since ChemicalSubstance is not in Biolink 2.0 but there are results
BioLink 2.0 was released two days ago and I don't think anyone implements it yet. Updates to BioLink 2.0 in ARAs and KPs is scheduled for July, I think.

**Comment by cbizon** [comment]
Yes, @edeutsch is correct, I would be very surprised if anybody had jumped ahead to implement biolink 2.0 (and really nobody should because we need to coordinate the timing)

**Comment by MarkDWilliams** [comment]
> using curl, BTE retrieves 136 results (including methamphetamine but not Lisdexamfetamine / vyvanse)
> 
> ```
> curl --silent --location --request POST "https://api.bte.ncats.io/v1/query" --header "Content-Type: application/json" --data-raw "{
>     \"message\": {
>         \"query_graph\": {
>             \"nodes\": {
>                 \"n0\": {
>                     \"categories\": [
>                         \"biolink:ChemicalSubstance\"
>                     ]
>                 },
>                 \"n1\": {
>                     \"ids\": [
>                         \"HGNC:6284\"
>                     ],
>                     \"categories\":[\"biolink:Gene\"]
>                 }
>             },
>             \"edges\": {
>                 \"e0\": {
>                     \"subject\": \"n0\",
>                     \"object\": \"n1\"
>                 }
>             }
>         }
>     }
> }"  | jq '.message.results | length'
> ```
> 
> Anyone have any idea what it means when the results are n/a but the TRAPI validates successfully? @dkoslicki I can't find any more detailed error message through ARAX -- am I missing it anywhere? @MarkDWilliams, I was expecting to be able to see more from ARS at https://ars.transltr.io/ars/api/messages/ec030f44-d4d0-468c-b608-72226658e5a0, but that gives me a 500 error. Any pointers?

@andrewsu We're still running the 1.1 queries through the dev instance while we shake the bugs out of the ITRB deployed instance.  You can find the PK you're looking or here
https://ars-dev.transltr.io/ars/api/messages/ec030f44-d4d0-468c-b608-72226658e5a0

**Comment by sstemann** [comment]
thanks @edeutsch here's the run from the this morning:  cdfad6fa-c0b7-4e40-9df2-332b6cb8c972

![image](https://user-images.githubusercontent.com/38321826/120814538-faa7f700-c51c-11eb-9440-98b8e5cc6a44.png)

**Comment by MarkDWilliams** [comment]
Also @andrewsu There was an issue with BTE's registration in dev, but I've corrected it and it is now returning 136 results.  Apologies for the confusion
116d6bf8-004a-4de7-89ee-994fecab8c4e

**Comment by bill-baumgartner** [comment]
Text mining provider returns results if `PR:000001849` is added to the `ids` list for this query. Our use of Protein Ontology concepts to represent (conflate) genes/gene products is detailed here: https://github.com/NCATSTranslator/Text-Mining-Provider-Roadmap/issues/81. This came up yesterday during the mini-hackathon as well (https://github.com/NCATSTranslator/minihackathons/issues/23) and in past stand-ups (https://github.com/NCATSTranslator/testing/issues/66#issuecomment-850508658). This issue is being addressed by the Data Modeling group (https://github.com/NCATSTranslator/minihackathons/issues/12).

**Comment by vdancik** [comment]
The following query (id on subject node) works for MolePro:

```
{
    "message": {
        "query_graph": {
            "nodes": {
                "n0": {
                    "categories": [
                        "biolink:ChemicalSubstance"
                    ]
                },
                "n1": {
                    "ids": [
                        "HGNC:6284"
                    ],
                    "categories":["biolink:Gene"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "n1",
                    "object": "n0"
                }
            }
        }
    }
}
```

**Comment by colleenXu** [comment]
Following up on Mark's earlier comment, BTE is returning results for this query. This is what I see using Mark's PK:

![Screen Shot 2021-06-09 at 10 58 03 PM](https://user-images.githubusercontent.com/43731687/121472594-4112b100-c976-11eb-9011-8a33b34729fd.png)
